conduct/COC_PATCHING_SCRIPT.md

```diff
@@ -0,0 +1,100 @@
+# Code of Conduct Patching Script
+
+This document explains the knowledge behind the patching script used
+to make any changes on the `CODE_OF_CONDUCT.md` file.
+
+The patching script lives under [`pkgjs/patch-my-code-of-conduct`]
+(http://github.com/pkgjs/patch-my-code-of-conduct) repository.
+It is developed as a GitHub workflow to make it easier for projects
+to adopt into their existing repositories.
+
+## Goal
+
+The goal of the patching script is to make Code of Conduct upgrade
+and review process easier to maintain for open-source projects.
+It removes any direct changes done to the original document, and
+provides an easier to review and accept changes to the document
+provided by the Contributor Covenant.
+
+## GitHub Workflow configuration
+
+The configuration file for the patching script lives inside
+`.github/workflows/patch-code-of-conduct.yml` file. The configuration
+file also includes an additional GitHub workflow to automatically
+open a pull request in the event of a changed file after the
+work of the patching scripts job is complete.
+
+### Settings
+
+- `base_url`: Base URL is the path to the preamble file (the file
+  that is prepended to the Code of Conduct document).
+- `template_url`: It is the URL to the Contributor Covenant website
+  that includes the original unchanged document.
+- `patch_file_path`: It is the path to the patching file.
+- `output_file_path`: It is the path to the file where the changes
+  should apply to. 
+
+### Files
+
+- `workflows/patch-code-of-conduct.yml`: This file contains
+  the necessary configuration to setup the patching script.
+- `CODE_OF_CONDUCT_PREAMBLE.md`: This file contains the text
+  prepended to the original Code of Conduct file. It may contain
+  project/organization specific information that may live outside
+  of the definition of the original Code of Conduct versions.
+- `CODE_OF_CONDUCT_PATCH.patch`: This file contains a patch
+  generated by the contributors of the project to update **only**
+  the texts defined in the Code of Conduct URL defined by the
+  patching script.
+
+## Workflows
+
+### Make a change to `PREAMBLE.md` file
+
+1. Open a pull-request updating either `CODE_OF_CONDUCT_PATCH.patch`
+   or `CODE_OF_CONDUCT_PREAMBLE.md` file. The default procedures of
+   merging a pull request will be followed as usual.
+2. Upon successful merging the proposed changes to the main branch
+   patching script will open a pull request with a target of `main`
+   branch updating `CODE_OF_CONDUCT.md` file with the proposed changes.
+3. Since the approval for the original change is done in step 1, unless
+   there is a bug with the CoC patching script this pull request opened
+   by the bot from the workflow should land without issue.
```

Comment:
*after* we've finished landing all the in-flight CoC PRs, I'd prefer we skip the second PR and make that an autocommit/push, to avoid confusion.

Reply:
In the long term, that is the correct approach, but first, I firmly believe we should validate if this approach has no underlying flaws.

Reply:
Totally agree.